/*
 * @begin:license
 *
 * Copyright (c) 2015-2018, Benjamin Niemann <pink@odahoda.de>
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * @end:license
 */

syntax = "proto2";

import "noisicaa/audioproc/public/player_state.proto";

package noisicaa.pb;

message NodeStateChange {
  required string realm = 1;
  required string node_id = 2;

  enum State {
    INACTIVE = 1;
    SETUP = 2;
    RUNNING = 3;
    BROKEN = 4;
    CLEANUP = 5;
  }
  required State state = 3;
}

message EngineStateChange {
  enum State {
    STOPPED = 1;
    SETUP = 2;
    RUNNING = 3;
    CLEANUP = 4;
  }
  required State state = 1;
  optional float load = 2;
}

message EngineNotification {
  optional PlayerState player_state = 1;
  repeated NodeStateChange node_state_changes = 2;
  repeated EngineStateChange engine_state_changes = 3;
  optional bytes perf_stats = 4;
}
